### ISSUE TITLE
Note To Self is listed under the wrong letter

### ISSUE THREAD

**Issue by Herohtar** [signalapp/Signal-Android#11322]
- [x] I have searched open and closed issues for duplicates
- [x] I am submitting a bug report for existing functionality that does not work as intended
- [x] I have read https://github.com/signalapp/Signal-Android/wiki/Submitting-useful-bug-reports
- [x] This isn't a feature request or a discussion topic

----------------------------------------

### Bug description
Note To Self is listed in the "wrong" place in the contact list.

### Steps to reproduce
1. Add yourself as a contact in your phone's address book
2. Give Signal access to phone's contacts
3. Open Signal and tap the pencil icon
4. Look for "Note To Self"

**Actual result:** "Note To Self" will be listed in the position it would normally be displayed in if it were a regular contact -- eg, if your name is "Herohtar" it will show up under the "H" section in the appropriate alphabetical ordering.
**Expected result:** "Note To Self" should show up under "N", or at the very top as a special conversation option.

### Device info

**Device:** Pixel 5
**Android version:** 11
**Signal version:** 5.10.8

**Comment by Kruemmelspalter** [comment]
For me, it's under N (my name doesn't begin with N)

**Comment by Sgn-32** [comment]
For example "Note To Self" is translated as "Shënim për Veten" on albanian. Then "Note To Self" should be under "S".

**Comment by Herohtar** [comment]
Yes, Note To Self is still listed in the wrong place in the contact list.

**Comment by Herohtar** [comment]
Yep, still a thing.

**Comment by Herohtar** [comment]
Definitely still a thing.
